### PR TITLE
Add naming format

### PR DESCRIPTION
**Pull request by y-myajima**
# Description

_description_with_details_and_reasoning_

## Assignees
- [x] `Assignees` is set

## Type of changes
- [ ] Apply one or more `labels` of the following that fits:
  - `bug`: Bug fix
  - `dependencies`: Dependency upgrades
  - `documentation`: Documentation changes
  - `enhancement`: New Feature
  - `good first issue`: First contribution
  - `logging`: Log changes
  - `refactor`: Refactoring (no functional changes, no api changes)

### Flags
```
- [ ] Breaks backward compatibility
- [ ] Requires a documentation update
- [ ] Has untestable code
```
---

## Checklist
```
- [ ] Followed the guidelines in the CONTRIBUTING document
- [ ] Added prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]` in the PR title if necessary
- [ ] Tested and linted the code
- [ ] Commented the code
- [ ] Made corresponding changes to the documentation
```

## Checklist for maintainer
```
- [ ] Use `Squash and merge`
- [ ] Double-confirm the merge message has prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]`
- [ ] Delete the branch after merge
```

### CODE REVIEW

File: pkg/util/directory_creator.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/AthenZ/k8s-athenz-sia/v3/third_party/log"
+)
+
+// CreateDirectoryFromOutPath creates the necessary directories for file output from the specified file path for the output token or certificate.
+func CreateDirectoryFromOutPath(outPath string) error {
```
Comment (mlajkim) [inline pkg/util/directory_creator.go:26]: Shorter name?
```suggestion
func CreateDirectory(path string) error {
```

File: pkg/util/directory_creator.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// Copyright 2023 LY Corporation
```
Comment (mlajkim) [inline pkg/util/directory_creator.go:1]: Fix the file name to
directory-creator.go
no `_` unless it is for test like: `directory-creator_test.go`

File: pkg/util/directory_creator.go (hunk 1 above)
Comment (mlajkim) [reply pkg/util/directory_creator.go:26]: Because file name `directory-creator.go`?

File: pkg/util/path_generator.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+package util
+
+import (
+	"fmt"
+	"html/template"
+	"strings"
+)
+
+// GeneratePath generates the output path for the credentials by applying the domain name, role name, and delimiter to a specified naming format.
+func GeneratePath(namingFormat, domain, role, delimiter string) (string, error) {
```
Comment (mlajkim) [inline pkg/util/path_generator.go:24]: this is cool

Comment (mlajkim) [reply pkg/util/path_generator.go:24]: Write some samples as comments of what kind of namingFormat is expected.

Comment (mlajkim) [reply pkg/util/path_generator.go:24]:
```suggestion
// GeneratePath generates the output path for the credentials by applying the domain name, role name, and delimiter to a specified naming format.
// i.e) namingFormat=="/var/run/athenz/rolecerts/{domain}{delimiter}{role}"
func GeneratePath(namingFormat, domain, role, delimiter string) (string, error) {
```

File: pkg/util/path_generator.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	if namingFormat == "" {
+		return "", fmt.Errorf("naming format is empty")
+	}
+	if domain == "" {
+		return "", fmt.Errorf("domain is empty")
+	}
+	// If the role is an empty string, the delimiter used to separate the domain name and the role name is not necessary.
+	if role == "" {
+		delimiter = ""
+	}
```
Comment (mlajkim) [inline pkg/util/path_generator.go:34]: Some sample comments for easier understanding?
```suggestion
	// If the role is an empty string, the delimiter used to separate the domain name and the role name is not necessary.
	// i.e) domain="athenz", role="" => User wants to fetch tokens directly associated to the domain.
	if role == "" {
		delimiter = ""
	}
```

**Review by mlajkim — COMMENTED** [review]
quick review

File: pkg/util/directory_creator.go (hunk 2 above)
Comment (mlajkim) [reply pkg/util/directory_creator.go:1]: directory generator? as well

**Comment by y-myajima** [comment]
Move to #140

**Pull request closed without merging**